lib/block-patterns.php

```diff
@@ -236,6 +236,58 @@ function load_remote_patterns() {
 	}
 }
 
+/**
+ * Register featured patterns from wordpress.org/patterns.
+ *
+ * @since 5.9.0
+ */
+function load_remote_featured_patterns() {
+	/**
+	 * Filter to disable remote block patterns.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param bool $should_load_remote
+	 */
+	$should_load_remote = apply_filters( 'should_load_remote_block_patterns', true );
+
+	if ( ! $should_load_remote ) {
+		return;
+	}
+
+	// TODO: check about the `featured` name.. Should this be reserved?
+	// Also check core logic: https://developer.wordpress.org/reference/classes/wp_block_pattern_categories_registry/
+	// Do we just allow collisions/overide with categories when they have the same name? Needs checking..
+	if ( ! WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'Featured' ) ) {
+		register_block_pattern_category( 'featured', array( 'label' => __( 'Featured', 'gutenberg' ) ) );
+	}
+
+	$request = new WP_REST_Request( 'GET', '/wp/v2/pattern-directory/patterns' );
+	// TODO: change this with new category id when created in patterns directory..
+	$request->set_param( 'category', 4 ); // temp gallery category id.
+
+	// TODO: check to update the controller here: https://github.com/WordPress/gutenberg/blob/trunk/lib/class-wp-rest-pattern-directory-controller.php#L321
+	// because it unsets the `per_page` param.
```

Comment:
I think this needs to change from unsetting entirely, to support the params and set defaults or make the other request to pass `-1` to fetch all. The API needs to support other use cases as well besides the pattern directory UI.

>Is there a reason to request just 2 patterns here?

`2` was just for demo and I guess we will show a bit more - maybe 10 or something? I believe we have to have a limit though because of performance and the current UI for selecting patterns, which is not super friendly for previewing a large number of patterns. We can revisit of course later..

Reply:
Turns out I was misremembering slightly - the wp.org API does not support pagination, [it always requests 100 patterns](https://meta.trac.wordpress.org/browser/sites/trunk/api.wordpress.org/public_html/patterns/1.0/index.php#L73). If/when we revamp the pattern inserter, we can update the API to support pagination, and then we can reenable these params here. For now, the featured category is manually curated, so you can be sure it's a smaller list.